### PR TITLE
Automated testing for galasactl - Issue 1413

### PR DESCRIPTION
**Pull request by jadecarino**
- Call the testing scripts in the CLI repo to test the built binaries of galasactl and various commands (tests for different binaries other than linux-amd64 to be added later, just a copy and paste changing the binary name really)
- Custom image created with Java, Maven and Gradle capabilities to execute the scripts
- Add the build of this custom image to the automation pipelines

Signed-off-by: Jade Carino <carino_jade@yahoo.co.uk>

### CODE REVIEW

**Comment by jadecarino** [comment]
Don't merge unless https://github.com/galasa-dev/cli/pull/71 is merged first.

**Comment by galasa-team** [comment]
Build successful

**Comment by galasa-team** [comment]
Build successful

**Comment by galasa-team** [comment]
Build successful

**Review by techcobweb — DISMISSED** [review]
Approved, though I did make a comment which may make you want to change a few things.

**Comment by galasa-team** [comment]
Build successful

File: pipelines/pipelines/cli/build-branch.yaml [hunk 1]
```diff
@@ -233,8 +233,127 @@ spec:
...
+      - chmod-local-script
+    params:
+    - name: context
+      value: $(context.pipelineRun.name)/cli
+    - name: image
+      value: docker.io/library/maven:3.8.6-openjdk-11
+    - name: command
+      value: 
+        - ./test-galasactl-local.sh
+        - --binary
```
[the remark this answers is not in the document]
Comment (jadecarino) [reply pipelines/pipelines/cli/build-branch.yaml]: Addressed in this PR: https://github.com/galasa-dev/cli/pull/78

**Review by techcobweb — APPROVED** [review]
Looks good.

**Pull request merged**